[FEATURE] Added Aurorae (KWin) Window Decoration Support

ItachiSan opened the pull request:
Add support for KWin Window Decoration system.
This plugin uses the  binary "plasma-apply-aurorae" in order to function.

ItachiSan committed:
feat: KWin window decoration support

Add support for KWin Window Decoration system.
This plugin uses the  binary "plasma-apply-aurorae" in order to function.

MartinVonReichenberg commented:
_Will be tested and eventually passed until Wednesday . . ._

ItachiSan commented:
Wednesday is here :wink:

MartinVonReichenberg commented on CMakeLists.txt:
```diff
@@ -59,7 +59,7 @@ file(WRITE "src/plugins/gtk.cpp" "${GTK_CONTENT}")
 # Install files not being a part of INSTALL TARGETS section
 install(FILES  ${CMAKE_CURRENT_BINARY_DIR}/dev.baduhai.Koi.xml     DESTINATION ${DBUS_INTERFACES_INSTALL_DIR})
 install(FILES  src/koi.desktop                                     DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/applications
-        RENAME local.KoiDbusInterface.desktop)
```
The name for the application's desktop file **must remain** as: `local.KoiDbusInterface.desktop` otherwise it **throws error** when launching Koi from KDE Plasma Desktop MENU. 
Or something else must be renamed/changed/adjusted inside the bundled DBus interface itself - I have NO idea what exactly . . .

See the screenshot when launching Koi from GUI:
<img width="418" height="134" alt="image" src="https://github.com/user-attachments/assets/aa69b30e-9981-4ee8-bc9c-b41bfc23ad25" />


**_Could you please test your Pull Requests more thoroughly!? Because this the third time you submit a PR with certain errors/mismatches or other invalidity which require adjustments - sorry, but this is a primary reason for me not "hopping" straight away into your submitted PRs . . ._** 

Please revert this line OR troubleshoot the insufficiency within the DBus interface itself. Thanks !


Thank you for compliance.
Martin from Liberec

ItachiSan replied:
Hello, I understand how you could get frustrated, however there is no "good practice" defined in this repo for how to test.
My way of testing is closing "regular" Koi and run the one in the source tree with a debugger (VSCode + GDB).

For next contributions, I will build a supported package (Arch) pointing to my source tree for validation. That should cover a more "realistic" use case.

For now I will revert this line as requested and perhaps will figure out how to get it to work in a second PR.

I am just giving my bits back to an app that is useful for me; I am doing it in my free time and I am no QT expert.
Again, I understand the frustration, but I would prefer moving forward a more constructive that passive-aggressive approach.

Bests,
Giovanni

MartinVonReichenberg replied:
By the way,

This is not about me being frustrated or related things to that this is about principle(s). If one makes changes to the source code of a graphical-based program and one changes something that is related to actually launching the program (solely via GUI) it also needs to be tested if everything works the same way _(or better better way)_ than before - that is pretty logical to anybody who have particular basics about open-source code.
In addition I am unsure how particularly undefined test specifications in this repository are when it comes unusable (non-launchable) Koi .DESKTOP file or having LIGHT Profiles under DARK Functions and vice-versa, _BUT never mind - it is not important at the moment . . ._

The thing is nobody asks you to contribute, and I am much more inferior to coding Linux / Unix bluff than you (obviously) so any effort is appreciated (even though is not appreciated by nobody specific - "generally writing").

The point is - _"Sometimes it is better to do nothing than mess up tens/hundreds/thousands things."_ - Yet this still happens nowadays among masses - Minority _(Corporations/Companies/Firms/Rich Individuals, ...)_ dictates/gives direction/provides leadership/set orders, but the development/direction-heading is often undesirable leaving the masses upset/hopeless/suffering in pain/ending in vain . . .
_**It all "boils" down to the same thing . . ."**_

MartinVonReichenberg reviewed (changes requested):
Changes required, see attached comment on line no. `62` at `CMakeLists.txt`

Otherwise, I am fine with the rest functionality-wise . . .

ItachiSan committed:
fix: Revert Desktop file renaming

Messes up with start up from Plasma menu.

Also, add Git exclusion for builds folder created from QT extension in VSCode.

MartinVonReichenberg reviewed (commented):
Please check the Aurorae button on the back side (Preferences) of `mainwindow.ui` - it does not Save / Apply changes after DESELECTING the Aurorae button (check-box); see attachments:

<img width="740" height="123" alt="Image" src="https://github.com/user-attachments/assets/6c468932-1293-47d0-a759-66316f62d4a4" />

<img width="789" height="349" alt="Image" src="https://github.com/user-attachments/assets/7b794e68-db1f-49d8-a78c-5f33df217c42" />

It reverts back to ENABLED.

The only way to disable it is manually via the `koirc` file:
```
[Aurorae]
enabled=false
```

Please troubleshoot and resolve,
Thank you!

ItachiSan commented:
Hi, thanks for the feedback!
I will make sure to reproduce the issue and debug it; I believe I will have some time tomorrow :smile:

ItachiSan committed:
fix: Aurorae checkbox toggling

- Hook preference savings to the correct checkbox
- Additionally clean empty strings from theme selection

MartinVonReichenberg reviewed (approved):
Ohh, congrats !

Good job!
Thank you!


_Fine by me . . ._
